include/dd/Package.hpp

```diff
@@ -1710,6 +1755,182 @@ namespace dd {
+            if (node.isTerminal()) {
+                auto amp = cn.getTemporary();
+                dd::ComplexNumbers::mul(amp, amplitude, node.w);
+                idx <<= level;
+                for (std::size_t i = 0; i < (1UL << level); i++) {
+                    amplitudes[idx++] = dd::ComplexValue{dd::ComplexTable<>::Entry::val(amp.r), dd::ComplexTable<>::Entry::val(amp.i)};
+                }
+
+                return;
+            }
+
+            auto a = cn.mulCached(amplitude, node.w);
+            exportAmplitudesRec(node.p->e[0], amplitudes, a, level - 1, idx << 1);
+            exportAmplitudesRec(node.p->e[1], amplitudes, a, level - 1, (idx << 1) | 1);
+            cn.returnToCache(a);
+        }
+        void exportAmplitudes(const dd::Package::vEdge& node, std::vector<ComplexValue>& amplitudes, dd::QubitCount nq) {
+            if (node.isTerminal()) {
+                // TODO special treatment
+                return;
+            }
+            auto weight = cn.getCached(1., 0.);
+            exportAmplitudesRec(node, amplitudes, weight, nq, 0);
+            cn.returnToCache(weight);
+        }
+
+        void addAmplitudesRec(const dd::Package::vEdge& node, std::vector<ComplexValue>& amplitudes, ComplexValue& amplitude, dd::QubitCount level, std::size_t idx) {
+            auto         ar = dd::ComplexTable<>::Entry::val(node.w.r);
+            auto         ai = dd::ComplexTable<>::Entry::val(node.w.i);
+            ComplexValue amp{ar * amplitude.r - ai * amplitude.i, ar * amplitude.i + ai * amplitude.r};
+
+            if (node.isTerminal()) {
+                idx <<= level;
+                for (std::size_t i = 0; i < (1UL << level); i++) {
+                    auto temp         = dd::ComplexValue{amp.r + amplitudes[idx].r, amp.i + amplitudes[idx].i};
+                    amplitudes[idx++] = temp;
+                }
+
+                return;
+            }
+
+            addAmplitudesRec(node.p->e[0], amplitudes, amp, level - 1, idx << 1);
+            addAmplitudesRec(node.p->e[1], amplitudes, amp, level - 1, idx << 1 | 1);
+        }
+        void addAmplitudes(const dd::Package::vEdge& node, std::vector<ComplexValue>& amplitudes, dd::QubitCount nq) {
+            if (node.isTerminal()) {
+                // TODO special treatment
+                return;
+            }
+            ComplexValue a{1., 0.};
+            addAmplitudesRec(node, amplitudes, a, nq, 0);
+        }
+
+        // transfers a decision diagram from another package to this package
+        template<class Edge>
+        Edge transfer(Edge& original) {
+            // POST ORDER TRAVERSAL USING ONE STACK   https://www.geeksforgeeks.org/iterative-postorder-traversal-using-stack/
+            Edge              root{};
+            std::stack<Edge*> stack;
+
+            std::unordered_map<decltype(original.p), decltype(original.p)> mapped_node{};
+
+            Edge* node = &original;
```

Comment:
A pointer to an `Edge` should not be called `node`. Maybe `current_edge`?

Reply:
`currentEdge` seems fine! I also adjusted the respective amplitude adding function parameters which were mistakenly named `node`.